feat(ble): Add the ability to enable and disable advertising (#2546)
zmkfirmware/zmk: docs/docs/keymaps/behaviors/bluetooth.md

| `BT_DISC`    | Disconnect from the 0-indexed profile by number, if it's currently connected and inactive; must include a number as an argument in the keymap to work correctly, e.g. `BT_DISC 0`. |
| `BT_ADV_OFF` | Disable advertising and disconnect from all profiles                                                                                                                               |
| `BT_ADV_ON`  | Enable advertising and attempt to reconnect to profiles                                                                                                                            |
| `BT_ADV_TOG` | Toggle advertising on and off                                                                                                                                                      |

:::note[Selected profile persistence]
The profile that is selected by the `BT_SEL`/`BT_PRV`/`BT_NXT` actions will be saved to flash storage and hence persist across restarts and firmware flashes.
However it will only be saved after [`CONFIG_ZMK_SETTINGS_SAVE_DEBOUNCE`](../../config/system.md#general) milliseconds in order to reduce potential wear on the flash memory.
:::

:::note[Advertising]

> Bluetooth advertising is enabled in all situations by default, and doesn't have to be explicitly enabled via the bindings above.

Reviewer: I would also add a comment here to state that it can be disabled by default.

| `BT_PRV`     | Switch to the previous profile, cycling through to the last one when the beginning is reached.                                                                                     |
| `BT_SEL`     | Select the 0-indexed profile by number; must include a number as an argument in the keymap to work correctly, e.g. `BT_SEL 0`.                                                     |
| `BT_DISC`    | Disconnect from the 0-indexed profile by number, if it's currently connected and inactive; must include a number as an argument in the keymap to work correctly, e.g. `BT_DISC 0`. |

> | `ADV_OFF`    | Disconnect from all profiles and disable advertising (The device will not show up as connectable on host devices)                                                                  |
> | `ADV_ON`     | Enable Bluetooth advertising, the device will show up as connectable                                                                                                               |

Reviewer 2: ```suggestion
| `ADV_OFF`    | Disable advertising and disconnect from all profiles                                                                |
| `ADV_ON`     | Enable advertising and attempt to reconnect to profiles                                                                                                   |
```
Maybe like this?

Author: Yep i think that's better, thank you

| `ADV_TOG`    | Toggle advertising on and off                                                                                                                                                      |

> 

Reviewer 3: I think it might be worth adding a small note here, saying that advertising is on by default.

I see this pattern in some users where they assume they need to set a setting for something to work, just because they see it exists.
